src/MOCKernel.h

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file MOCKernel.h
+ * @brief An MOCKernel object
+ * @date May 5, 2015
+ * @author Geoffrey Gunow, MIT, Course 22 (geogunow@mit.edu)
+ */
+
+#ifndef MOCKERNEL_H_
+#define MOCKERNEL_H_
+
+#include "Python.h"
+#include "Track.h"
+#include "Geometry.h"
+#include "Quadrature.h"
+
+
+/* Forward declaration of TrackGenerator */
+class TrackGenerator;
+
+
+/**
+ * @class MOCKernel MOCKernel.h "src/MOCKernel.h"
+ * @brief An MOCKernel object specifies a functionality to apply to MOC
+ *        segments.
+ * @details An MOCKernel is an object that owns some specified data and
+ *          contains an "execute" function which applies some functionality
+ *          to the data. This is useful in MOC where it is very common to
+ *          apply some function to segment data in either a nested loop
+ *          structure or from on-the-fly calculations. Kernels specify the
+ *          actions applied to the segments, reducing the need for repeated
+ *          code. This class is the parent class of CounterKernel and
+ *          VolumeKernel. A generic MOCKernel should not be explicity
+ *          instantiated. Instead, an inhereting class should
+ *          be instantiated which describes the "execute" function.
+ */
+class MOCKernel {
+
+protected:
+
+  /** Count referring to the segment number */
+  int _count;
+
+  /** Maximum optical path length when forming segments */
+  FP_PRECISION _max_tau;
+
+public:
+
+  MOCKernel(TrackGenerator* track_generator, int row_num);
+  virtual ~MOCKernel();
+
+  /* Function to get the current segment count */
+  int getCount();
+
+  /* Sets the max optical path length to a different value */
+  void setMaxOpticalLength(FP_PRECISION max_tau);
+
+  /* Prepare MOCKernel for handling a new track */
+  virtual void newTrack(Track* track);
+
+  /* Executing function describes kernel behavior */
+  virtual void execute(FP_PRECISION length, Material* mat, int id,
+      int cmfd_surface_fwd, int cmfd_surface_bwd)=0;
+
+};
+
+
+/**
+ * @class CounterKernel MOCKernel.h "src/MOCKernel.h"
+ * @brief Counts the number of segments of a track
+ * @details A CounterKernel inherets from MOCKernel and is a kernel which
+ *          counts the number of segments in a track by incrementing the
+ *          _count variable by the number of legitimate segment lengths
+ *          (less than the max optical path length) in the input length.
+ */
+class CounterKernel: public MOCKernel {
+public:
```

Comment:
I believe what he means is that there should be a blank line between `class CounterKernel : public MOCKernel {` and `public`. Take a look at the other header files to see.